generate-stackbrew-library.sh

```diff
@@ -1,59 +1,26 @@
 #!/usr/bin/env bash
-set -eu
-
-declare -A aliases=(
-	[6]='latest'
-)
-defaultVariant='debian'
+set -Eeuo pipefail
 
 self="$(basename "$BASH_SOURCE")"
 cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
 
-versions=( */ )
-versions=( "${versions[@]%/}" )
-
-# sort version numbers with highest first
-IFS=$'\n'; versions=( $(echo "${versions[*]}" | sort -rV) ); unset IFS
-
 # get the most recent commit which modified any of "$@"
 fileCommit() {
 	git log -1 --format='format:%H' HEAD -- "$@"
 }
 
-# get the most recent commit which modified "$1/Dockerfile" or any file COPY'd from "$1/Dockerfile"
-dirCommit() {
-	local dir="$1"; shift
-	(
-		cd "$dir"
-		fileCommit \
-			Dockerfile \
-			$(git show HEAD:./Dockerfile | awk '
-				toupper($1) == "COPY" {
-					for (i = 2; i < NF; i++) {
-						print $i
-					}
-				}
-			')
-	)
-}
-
-getArches() {
+getArchesJson() {
 	local repo="$1"; shift
-	local officialImagesBase="${BASHBREW_LIBRARY:-https://github.com/docker-library/official-images/raw/HEAD/library}/"
-
-	local parentRepoToArchesStr
-	parentRepoToArchesStr="$(
-		find -name 'Dockerfile' -exec awk -v officialImagesBase="$officialImagesBase" '
-				toupper($1) == "FROM" && $2 !~ /^('"$repo"'|scratch|.*\/.*)(:|$)/ {
-					printf "%s%s\n", officialImagesBase, $2
-				}
-			' '{}' + \
-			| sort -u \
-			| xargs -r bashbrew cat --format '["{{ .RepoName }}:{{ .TagName }}"]="{{ join " " .TagEntry.Architectures }}"'
-	)"
-	eval "declare -g -A parentRepoToArches=( $parentRepoToArchesStr )"
+	: "${BASHBREW_LIBRARY:-https://github.com/docker-library/official-images/raw/HEAD/library}/"
+
+	jq --raw-output '.[].variants[].from' versions.json \
+		| sort -u \
+		| xargs -r bashbrew cat --format '{ "{{ .RepoName }}:{{ .TagName }}": {{ json .TagEntry.Architectures }} }' \
+		| jq --compact-output --null-input '[ inputs ] | add'
```

Comment:
This is just `slurp` :sweat_smile:

```suggestion
		| jq --compact-output --slurp 'add'
```